src/node.js

```diff
@@ -390,6 +391,78 @@
     }
   };
 
+  startup.processPromises = function() {
+    // this should be kept in sync with PromiseRejectEvent in v8.h
+    var kPromiseRejectWithNoHandler = 0;
+    var kPromiseHandlerAddedAfterReject = 1;
+
+    var timers = NativeModule.require('timers');
+    var unhandledPromises = new WeakMap();
+    // A hack to work around uniterable WeakMaps.
+    var unhandledPromiseKeys = [];
+    var fullTurnQueued = false;
+
+    function queueFullTurn() {
+      if (!fullTurnQueued) {
+        fullTurnQueued = true;
+        timers.setImmediate(function() {
+          process.nextTick(afterFullTurn);
```

Comment:
@domenic so so you think that `setImmediate` is the way to go here or should something else be considered?

What about using `Promise.resolve().then(...)` instead.

@vkurchatkin what would guarantee one turn has passed but nothing more? Do you have an idea for a more precise method to check this?

Adding a link to our discussion in StackOverflow which also contains links to relevant threads for everyone's benefit:  http://stackoverflow.com/questions/27647742/promise-resolve-then-vs-setimmediate-vs-nexttick?lq=1